packages/eclipse/src/components/inline-toc.tsx

```
@@ -14,30 +15,80 @@ export interface InlineTocProps extends ComponentProps<"div"> {
+    observerRef.current = new IntersectionObserver(
+      (entries) => {
+        const visible = entries
+          .filter((e) => e.isIntersecting)
+          .sort((a, b) => a.boundingClientRect.top - b.boundingClientRect.top);
+
+        if (visible.length > 0) setActiveId(visible[0].target.id);
+      },
+      { rootMargin: "0px 0px -60% 0px", threshold: 0 },
+    );
+
+    elements.forEach((el) => observerRef.current!.observe(el));
```

Biome (2.4.4)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)